Assets/HoloToolkit/Input/Scripts/Utilities/ControllerFinder.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+#if UNITY_2017_2_OR_NEWER
+using UnityEngine.XR.WSA.Input;
+#else
+using UnityEngine.VR.WSA.Input;
+#endif
+
+namespace HoloToolkit.Unity.InputModule
+{
+    /// <summary>
+    /// ControllerFinder is a base class providing simple event handling for getting/releasing MotionController Transforms
+    /// </summary>
+    public class ControllerFinder : MonoBehaviour
+    {
+        #region public members
+        public MotionControllerInfo.ControllerElementEnum Element { get { return element; } }
+        public Transform ElementTransform { get; private set; }
+        public bool IsAttached { get { return isAttached; } set { isAttached = value; } }
```

Comment:
Should `IsAttached` be implemented here? This seems like something that makes sense in `AttachToController`, but not necessarily in this more generic case.